e2e/applied-polyfill-php80-on-php80-sets/rector.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+use Rector\Config\RectorConfig;
+use Rector\ValueObject\PhpVersion;
+
+return RectorConfig::configure()
+    ->withPaths([__DIR__ . '/src'])
+    ->withPhpVersion(PhpVersion::PHP_74) // using php 7.4
+    ->withPhpSets(php80: true); // enable php 8.0 set, it will be applied as there is polyfill
```

Comment:
will the php8 set introduce things which require php8 syntax and cannot be polyfilled?

Reply:
Polyfill rules are based on `RelatedPolyfillInterface`, if the rules are not implements `RelatedPolyfillInterface`, it will only check by php feature, see 

https://github.com/rectorphp/rector-src/blob/483849a4344ab480a630f1263a76ca063a57b172/src/VersionBonding/PhpVersionedFilter.php#L31-L38

which on some rule; eg switch to match not implements, see

https://github.com/rectorphp/rector-src/blob/61cb7b5042ebca12c18e1f2df22ecf3ba9d26bdb/rules/Php80/Rector/Switch_/ChangeSwitchToMatchRector.php#L33

And on strpos to str contains, it implements, see 

https://github.com/rectorphp/rector-src/blob/61cb7b5042ebca12c18e1f2df22ecf3ba9d26bdb/rules/Php80/Rector/Identical/StrStartsWithRector.php#L28